src/python/ensembl/utils/argparse.py

```diff
@@ -0,0 +1,129 @@
+        if parent_path.exists() and not os.access(parent_path, os.W_OK):
+            parser.error(f"'{dst_path}' is not writable")
+    return dst_path
+
+
+class ArgumentParser(argparse.ArgumentParser):
+    """Extends :class:`argparse.ArgumentParser` with additional methods and functionality.
+    
+    The default behaviour of the help text will be to display the default values on every non-required
+    argument, i.e. optional arguments with "required=False".
+
+    """
+
+    def __init__(self, *args, **kwargs) -> None:
+        """Extends the base class to include the information about default argument values by default."""
+        super().__init__(*args, **kwargs)
+        self.formatter_class = argparse.ArgumentDefaultsHelpFormatter
+
+    def add_argument(self, *args, **kwargs) -> None:
+        """Extends the parent function by excluding the default value in the help text when not provided.
+        
+        Only applied to required arguments without a default value, i.e. positional arguments or optional
+        arguments with "required=True".
+
+        """
+        if kwargs.get("required", False) and "default" not in kwargs:
+            kwargs["default"] = argparse.SUPPRESS
+        super().add_argument(*args, **kwargs)
+
+    def add_argument_src_path(self, *args, **kwargs) -> None:
+        """Adds :class:`pathlib.Path` argument, checking if it exists and it is readable at parsing time.
+
+        If "metavar" is not defined it is added with "PATH" as value to improve help text readability.
+
+        """
+        if "metavar" not in kwargs:
+            kwargs["metavar"] = "PATH"
+        kwargs.pop("type", None)
+        self.add_argument(*args, **kwargs, type=lambda x: _validate_src_path(self, x))
+
+    def add_argument_dst_path(self, *args, **kwargs) -> None:
+        """Adds :class:`pathlib.Path` argument, checking if it is writable at parsing time.
+
+        If "metavar" is not defined it is added with "PATH" as value to improve help text readability.
+
+        """
+        if "metavar" not in kwargs:
+            kwargs["metavar"] = "PATH"
+        kwargs.pop("type", None)
+        self.add_argument(*args, **kwargs, type=lambda x: _validate_dst_path(self, x))
+
+    def add_server_arguments(self, prefix: str = "") -> None:
+        """Adds the usual set of arguments needed to connect to a server.
+        
+        This means: "--host", "--port", "--user" and "--password" (optional).
+
+        Args:
+            prefix: Prefix to add the each argument, e.g. if prefix is "src_", the arguments will be
+                "--src_host", etc.
+
+        """
+        self.add_argument(f"--{prefix}host", required=True, help="Name of the host")
+        self.add_argument(f"--{prefix}port", required=True, type=int, help="Port number")
+        self.add_argument(f"--{prefix}user", required=True, help="User name")
+        self.add_argument(f"--{prefix}password", metavar="PWD", default="", help="Host password")
```

Comment:
It is a tricky one, particularly with those capital letters involved. Will try to see if there is something smart that can be done. If not, it may stay like this for now.